### ISSUE TITLE
Purpose of checkSignature.sh file (and other details)

### ISSUE THREAD

**Issue by smileBeda** [ozzi-/php-app-updater#1]
Hello @ozzi- from a fellow CH living abroad :)

This is a cool library that you have created here, and I have adopted it (with some changes and refactoring) to a project of mine.
I was however wondering what the purpose of `checkSignature.sh` file is?
I get it - it validates the signature. But we just created that signature _ourselves_ on _our server_ using _our_ zip file.
So why would we want to validate it?
The end user (client) wouldn't use this file anyway, so I am confused as of why we as the "owners" would use it.

Then out of curiosity I run it, and surprisingly it told me that the signature was invalid (something with "tabs incorrect")
I am pretty sure the signature is valid as I had just created it, and also the entire process works just fine.

Mind to shed some light?

---

**A few details**
- I noticed that `updateDB.sh` uses (probably) windows EOLs and throws errors when used (on unix systems). 
**Simple trick here was to resave the file using an IDE and set the EOLs to `unix` instead. Perhaps worth a note on the readme?**
- Further I thought it might be useful to the new users to get a heads up about file permissions on the server. By default, servers would save all files with public read permissions (644). In my case, being a bit paranoid, I used these permissions (all files owned by a **dedicated** user, not root):
`createKeyPair.sh 400`
`createSignature.sh 500`
`db.json 644`
`hash 600` (I amended the `createSignature.sh` file to include a `chmod 600 hash` on line to to achieve this automatically)
`index.php 644`
`private_key.pem 600`
`public_key.pem 644`
`signature 644`
`updateDB.sh 400` 
This seem to be the least required capabilities to have the system working.

Thanks for the work you did on this, it was a great kickstarter for me.

**Comment by ozzi-** [comment]
Hi Beda :)
The signature check helps against corruption of the data during download. But yes, I agree this is an optional step.
If we would place the signature on another server, it would serve as a protection mechanism if the download server is taken over by a malicious actor (unless the actor takes over both separate servers and replaces the signature for his malicious update).

Concerning the EOL & file permissions, I will gladly add this to the readme.

Glad to hear it helped you with your projects!
Happy coding 😃

**Comment by smileBeda** [comment]
Great, thank you!

I am still a bit confused about `checkSignature.sh` (its purpose is clear, but its results not).
When I run it, it always returns `Verification failure`
Pretty sure thou, the signature generated is correct, after all, _locally_ when pulling the ZIP down, it does the same operation (just with PHP) and there it does not return failure.

I understand the `checkSignature.sh` is there to verify the signature _placed on the server_ (which we then expose in the DB Json for comparison from local).
So I see no reason why the verification commands would fail.

Why I noticed is that it throws some errors about `rsautl` being deprecated, yet it works just fine to generate it, so I wouldn't see a reason why verification would fail.

I found [this interesting SO thread](https://stackoverflow.com/questions/40565364/generating-signing-and-verifying-digital-signature) where they seem to have similar issues.
I'll dig a bit into it as it would likely require to rewrite some of the logic locally too.

On your end that command runs fine?

**Comment by ozzi-** [comment]
Yeah with Git Bash for Windows I am running into the same error. I guess one would need to change the commands to something like:

createSignature.sh
```bash
openssl dgst -sha512 -sign private_key.pem -out signature repo/$1
openssl base64 -in signature -out signatureb64
cat signatureb64
```
